sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/util/PrivateFieldAccessHelper.java

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.implementation.util;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.lang.reflect.Field;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+/**
+ * The helper class for getter only models converters.
+ */
+public class PrivateFieldAccessHelper {
+    private static final ClientLogger LOGGER = new ClientLogger(PrivateFieldAccessHelper.class);
+
+    /**
+     * Set value to the model private properties.
+     *
+     * @param obj The instance which sets the value to.
+     * @param fieldName The fieldName to set.
+     * @param value The value sets to the instance
+     * @param <T> Generic type of models.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> void set(T obj, String fieldName, Object value) {
+        try {
+            Field f1 = obj.getClass().getDeclaredField(fieldName);
+            AccessController.doPrivileged(new PrivilegedAction<T>() {
```

Comment:
Should we check `Field.isAccessible` before needing to call into privileged code?

Reply:
This is a tradeoff.

I don't know if we want to block customer to reflect on the return model if they turn on the SecurityManager. The code here is to avoid failure on reflection when customer has SecuirityManager on their machines.

Reply:
Oh, I mean check if the field is already accessible before making the privileged call.